pkg/log/rate_limited_logger_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package log
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/time/rate"
+	"gotest.tools/v3/assert"
+)
+
+type logMessage struct {
+	Level   string `json:"level"`
+	Message string `json:"msg"`
+}
+
+func TestRateLimitedLog(t *testing.T) {
+	zapLogger, logFile := createZapTestLogger()
+	logger := getTestRateLimitedLog(zapLogger, time.Minute)
+
+	// this won't last over one minute, assert there is only one record in log file
+	for i := 0; i < 10000; i++ {
+		logger.Info("YuniKorn")
+	}
```

Comment:
I suggest a different approach here, we don't know how much time it takes to run 10000 iterations.

For example, invoke `logger.Info("Yunikorn")` in every 100ms for 2 seconds. That means 20 calls in total. Then you check the output.

Reply:
copy that !